### PR TITLE
Tickets/dm 34521

### PR DESCRIPTION
**Pull request by fjammes**

### CODE REVIEW

File: src/admin/python/lsst/qserv/admin/qservCli/launch.py [hunk 1]
```diff
@@ -1067,15 +1070,25 @@ def integration_test(
...
+    for opt, var in (
+        ("run-tests", run_tests),
+        ("compare-results", compare_results),
+    ):
+        if var:
+            prefix = "--"
+        else:
+            prefix = "--no-"
+            args.append(f"{prefix}{opt}")
+
```
Comment (n8pease) [inline src/admin/python/lsst/qserv/admin/qservCli/launch.py:1091]: this is cute but for clarity I think this is better
```
args.append("--run-tests" if run_tests else "--no-run-tests")
args.append("--compare-results" if run_tests else "--compare-results")
```

Comment (fjammes) [reply src/admin/python/lsst/qserv/admin/qservCli/launch.py:1091]: :+1:

File: src/admin/python/lsst/qserv/admin/itest_load.py [hunk 2]
```diff
@@ -487,6 +510,43 @@ def _get_cases(cases: Optional[List[str]], test_cases_data: List[Dict[Any, Any]]
...
+
+    Parameters
+    ----------
+    test_cases_data : `list` [ `dict` ]
+        Dicts whose values will be used to initialize a LoadDb class instance.
+
+    """
+    _log.debug("test cases data:\n%s", test_cases_data)
+    cases_data = _get_cases(None, test_cases_data)
+    work_dir = os.path.join("/tmp", "datasets")
```
Comment (andy-slac) [inline src/admin/python/lsst/qserv/admin/itest_load.py:527]: Why is `/tmp` hardcoded? Should it be configurable?

File: src/admin/python/lsst/qserv/admin/qservCli/launch.py [hunk 3]
```diff
@@ -1141,6 +1144,65 @@ def itest_rm(project: str, dry: bool) -> None:
...
 
 
+def prepare_data(
+    qserv_root: str,
+    itest_container: str,
+    qserv_image: str,
+    itest_file: str,
+    dry: bool,
+    project: str,
+
```
Comment (andy-slac) [inline src/admin/python/lsst/qserv/admin/qservCli/launch.py:1154]: Extra empty line

File: src/admin/python/lsst/qserv/admin/tests/test_qserv_cli.py [hunk 4]
```diff
@@ -178,6 +205,24 @@ def test_OptDefault_default(self, itest_mock):
...
+        self.assertEqual(res.exit_code, 0)
+        prepare_data_mock.assert_called_once()
+        expected = os.path.abspath(
+            # This is the location of the path *inside* the build container,
+            # because that's where the unit tests run. This is from the install
+            # location of the unit tests back up the tree to the root qserv
+            # folder.
+            os.path.join(__file__, "../../../../../../../..")
+        )
+        print(expected)
```
Comment (andy-slac) [inline src/admin/python/lsst/qserv/admin/tests/test_qserv_cli.py:223]: Do you need `print` in the test code?

**Review by andy-slac — APPROVED** [review]
Looks OK, few minor comments.

File: src/admin/python/lsst/qserv/admin/cli/entrypoint.py [hunk 5]
```diff
@@ -356,6 +358,22 @@ def integration_test(
...
 
 
+@entrypoint.command()
+@tests_yaml_option()
+def prepare_data(
+    tests_yaml: str,
+) -> None:
+    """Unzip and partition test datasets.
+
+    TESTS_YAML is the yaml file paths that contains connection information & describes tests to load and run.
```
[the remark this answers is not in the document]
Comment (fjammes) [reply src/admin/python/lsst/qserv/admin/cli/entrypoint.py:368]: :+1:

**Pull request merged**